planner/core/find_best_task.go

```diff
@@ -260,7 +393,7 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 			}, nil
 		}
 		if path.isTablePath {
-			tblTask, err := ds.convertToTableScan(prop, path)
+			tblTask, err := ds.convertToTableScan(prop, path, candidate.matchProp)
```

Comment:
It's better to refine `ds.convertToTableScan's` arguments to `(prop, candidate)` ?